File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.py
import math 

def solution(n):
    answer = -1
    x = int(math.sqrt(n)) #제곱근 구하기
    if x*x == n :
        answer = (x+1) ** 2 #제곱 구하기

    
    return answer

print(solution(121))
print(solution(3))

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

long long solution(long long n) {
    long long answer = 0;
    long long int_val = (long long)sqrt(n);
    
    
    if((int_val * int_val) == n) {
        answer = (int_val+1) * (int_val+1);
    } else {
        answer = -1;
    }
    return answer;
}